From 0bd6313149c04da62cf81e8fb11e8f48e9a38cb0 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Fri, 10 Jul 2020 12:00:15 +0200
Subject: [M85] Do not declare method by copy parameter using abstract type in
 NearbyConnectionManager

OnIncomingConnection virtual method of NearbyConnectionManager has an abstract parameter passed
by copy. This should not compile. Replace it with a pointer (as it is done in master as part
of another patch).

Bug: 819294
Change-Id: Ide3d8f28e9e60a0d74d7f62106013dc024006152
---
 chrome/browser/nearby_sharing/nearby_connections_manager.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/chrome/browser/nearby_sharing/nearby_connections_manager.h b/chrome/browser/nearby_sharing/nearby_connections_manager.h
index d6ac91e86815..f98f98badc67 100644
--- a/chrome/browser/nearby_sharing/nearby_connections_manager.h
+++ b/chrome/browser/nearby_sharing/nearby_connections_manager.h
@@ -27,9 +27,10 @@ class NearbyConnectionsManager {
   class IncomingConnectionListener {
     virtual ~IncomingConnectionListener() = default;
 
-    virtual void OnIncomingConnection(const std::string& endpoint_id,
-                                      const std::vector<uint8_t>& endpoint_info,
-                                      NearbyConnection connection) = 0;
+    virtual void OnIncomingConnection(
+        const std::string& endpoint_id,
+        const std::vector<uint8_t>& endpoint_info,
+        std::unique_ptr<NearbyConnection> connection) = 0;
   };
 
   // A callback for handling discovered devices while discovering.
-- 
2.20.1

